### PR TITLE
Fix proxy authentication failing on Mono for https sources

### PR DESCRIPTION
**Pull request by rohit21agrawal**
This fixes a workaround in the issue : https://github.com/NuGet/Home/issues/2992

For HTTPS sources, NuGet.exe was throwing a WebException running on Mono. This fix is basically a hack to determine if the runtime is Mono and if it returns a 407 error code, wait for credentials instead of throwing an exception.

CC: @alpaix @joelverhagen @mrward @rrelyea @emgarten

### CODE REVIEW

File: src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/ProxyAuthenticationHandler.cs [hunk 1]
```diff
@@ -103,6 +102,11 @@ public class ProxyAuthenticationHandler : DelegatingHandler
         // Returns true if the cause of the exception is proxy authentication failure
         private static bool ProxyAuthenticationRequired(Exception ex)
         {
+            if (RuntimeEnvironmentHelper.IsMono && !(ex is HttpRequestException))
```
Comment (joelverhagen) [inline src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/ProxyAuthenticationHandler.cs]: `&& ex is WebException`?

Comment (rohit21agrawal) [inline src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/ProxyAuthenticationHandler.cs]: fixed

File: src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/ProxyAuthenticationHandler.cs [hunk 2]
```diff
@@ -103,6 +102,12 @@ public class ProxyAuthenticationHandler : DelegatingHandler
         // Returns true if the cause of the exception is proxy authentication failure
         private static bool ProxyAuthenticationRequired(Exception ex)
         {
+            // HACK!!! : This is a hack to workaround Xamarin Bug 19594
+            if (RuntimeEnvironmentHelper.IsMono && (ex is WebException))
+            {
+                return IsMonoProxyAuthenticationRequiredError((WebException) ex);
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/ProxyAuthenticationHandler.cs]: nit: this should be cast one time, code analysis use to complain about this but I don't think that check is done in our current build

**Comment by emgarten** [comment]
LGTM :shipit:

**Comment by alpaix** [comment]
:shipit:

File: src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/ProxyAuthenticationHandler.cs (hunk 2 above)
Comment (rohit21agrawal) [inline src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/ProxyAuthenticationHandler.cs]: fixed

**Pull request merged**